### ISSUE TITLE
Key Error in CTC edge error computation with non-split vertices 

### ISSUE THREAD

**Issue by cmalinmayor** [Janelia-Trackathon-2023/traccuracy#137]
* traccuracy version: 0.1.0
* Python version: 3.11.6
* Operating System: macOS

# Description

I was running the CTC metrics on an example and kept getting exceptions. First, when the induced graph was empty (no TP nodes), the CTC metrics raised a confusing error. I fixed this in b365b8f330d077b3f465a0004882787893eda532, but still am getting a KeyError when looking up the expected computed edge endpoints in the gt_comp_mapping. I have created a minimal test case in 0802b84fa9fc9dce58a45ca37b44e27531890a99 that I believe replicates the issue.

My understanding of the issue is as follows: We are looking up the expected computed edge in the gt_comp_mapping, which was constructed to only include mappings where the computed node was included in the "induced graph". The “induced graph” only includes TP computed nodes, and not ones labeld Non-split. The if statement here 
https://github.com/Janelia-Trackathon-2023/traccuracy/blob/0802b84fa9fc9dce58a45ca37b44e27531890a99/src/traccuracy/track_errors/_ctc.py#L140-L145
only considers if the ground truth node is a false negative. If the ground truth node was matched to a non-split node, it is not labeled false negative, so we get past this if statement and try and identify the expected computed edge. However, the matched non-split computed node will not be in the induced graph and then not in the mapping, hence the key error. 

@DragaDoncila I need your guidance on how to fix this. Is a GT edge matched to a predicted edge where one end is a non-split vertex considered a False negative? Does the above explanation even make sense or is something else fishy going on here?

# Minimal example to reproduce the bug

https://github.com/Janelia-Trackathon-2023/traccuracy/blob/0802b84fa9fc9dce58a45ca37b44e27531890a99/tests/track_errors/test_ctc_errors.py#L139-L186

# Severity
- [x] Unusable
- [ ] Annoying, but still functional
- [ ] Very minor

**Comment by DragaDoncila** [comment]
@cmalinmayor just so you know, I am aware of this issue and am thinking it through and will get back to you. Most likely not until Monday of next week though, sorry!

**Comment by DragaDoncila** [comment]
@cmalinmayor yep it's a bug. From the original paper:

![image](https://github.com/Janelia-Trackathon-2023/traccuracy/assets/17995243/f0807320-414e-4102-8390-fe39ca193065)

So in the example you've constructed, the induced graph would be empty. We'd have 4 NS vertex errors (for the 2 GT vertices matched to each comp vertex) and two FN vertices. We'd then have 4 FN edges because there are no correct edges in our computed graph (according to the metrics). 

It's annoying, but I think we'll need to replace [these lines](https://github.com/Janelia-Trackathon-2023/traccuracy/blob/main/src/traccuracy/track_errors/_ctc.py#L144-L145) with

```python
        source_comp_id = gt_comp_mapping.get(source, None)
        target_comp_id = gt_comp_mapping.get(target, None)
```

And then before making the expected computed edge, check that they are both not `None`.

We could make a `WAS_NS` or similar attribute for GT nodes instead, but since the metric doesn't define them in that way, I don't know if I'd recommend it...

**Comment by cmalinmayor** [comment]
@DragaDoncila Thanks! I'll try a fix now based on the feedback. I think it might be in the IOU matcher too, I also got a KeyError there when I tried to run it on the same data, but I will have to take a closer look.

**Comment by cmalinmayor** [comment]
Closed by #141
